actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -313,12 +313,23 @@ def server_software
     def raw_post
       unless has_header? "RAW_POST_DATA"
         raw_post_body = body
-        set_header("RAW_POST_DATA", raw_post_body.read(content_length))
+        set_header("RAW_POST_DATA",
+          if chunked?
+            raw_post_body.rewind if raw_post_body.respond_to?(:rewind)
```

Comment:
for chunked requests had to rewind again before reading for some reason

Reply:
I commented out this line, and ran the tests in `test_case_test.rb` and they all passed.  Are you sure this rewind is necessary?  And if so, could a test be added to ensure it doesn't regress?

Reply:
The rewind is necessary, in the rails app I linked, if this line is commented out, it never worked.

Will test this out again and make sure it's working as it should (perhaps this test itself is not properly testing out the feature), will test and confirm.

Thanks for the reminder on this PR, been a while since I looked at it. 👍